Parse/Internal/Commands/CommandRunner/URLSession/Session/PFURLSession.h

```diff
@@ -17,15 +17,29 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class PFURLSession;
+
+@protocol PFURLSessionDelegate <NSObject>
+
+- (void)urlSession:(PFURLSession *)session willPerformURLRequest:(NSURLRequest *)request;
+- (void)urlSession:(PFURLSession *)session didPerformURLRequest:(NSURLRequest *)request withURLResponse:(nullable NSURLResponse *)response;
```

Comment:
If we fail below the protocol level (e.g. user switches wifi off), this will be called without a URL Response, and no way to get the error out. Thoughts on having a way to get the underlying network error if one exists here?